sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala

```diff
@@ -465,4 +465,35 @@ class PersistedViewTestSuite extends SQLViewTestSuite with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-36011: Disallow altering permanent views based on temporary views or UDFs") {
+    import testImplicits._
+    val table = "t"
+    withTable(table) {
+      (1 to 10).toDF("id").write.saveAsTable(table)
+      val view = "v1"
+      val tView = "v2"
```

Comment:
nit: I feel it's clearer to just hardcode "t", "v1", "v2"

Reply:
OK